charts/operator/values.go

```diff
@@ -0,0 +1,154 @@
+// +gotohelm:filename=_values.go.tpl
+package operator
+
+import (
+	_ "embed"
+
+	"github.com/invopop/jsonschema"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	//go:embed values.yaml
+	DefaultValuesYAML []byte
+
+	//go:embed values.schema.json
+	ValuesSchemaJSON []byte
+)
+
+type Values struct {
+	NameOverride            string                        `json:"nameOverride"`
+	FullnameOverride        string                        `json:"fullnameOverride"`
+	ReplicaCount            int32                         `json:"replicaCount"`
+	ClusterDomain           string                        `json:"clusterDomain"`
+	Image                   Image                         `json:"image"`
+	KubeRBACProxy           KubeRBACProxyConfig           `json:"kubeRbacProxy"`
+	Configurator            Image                         `json:"configurator"`
+	Config                  Config                        `json:"config"`
+	ImagePullSecrets        []corev1.LocalObjectReference `json:"imagePullSecrets"`
+	LogLevel                string                        `json:"logLevel"`
+	RBAC                    RBAC                          `json:"rbac"`
+	Webhook                 Webhook                       `json:"webhook"`
+	ServiceAccount          ServiceAccountConfig          `json:"serviceAccount"`
+	Resources               corev1.ResourceRequirements   `json:"resources"`
+	NodeSelector            map[string]string             `json:"nodeSelector"`
+	Tolerations             []corev1.Toleration           `json:"tolerations"`
+	Affinity                WrappedAffinity               `json:"affinity" jsonschema:"deprecated"`
+	Strategy                *appsv1.DeploymentStrategy    `json:"strategy,omitempty"`
+	Annotations             map[string]string             `json:"annotations,omitempty"`
+	PodAnnotations          map[string]string             `json:"podAnnotations"`
+	PodLabels               map[string]string             `json:"podLabels"`
+	AdditionalCmdFlags      []string                      `json:"additionalCmdFlags"`
+	CommonLabels            map[string]string             `json:"commonLabels"`
+	Monitoring              MonitoringConfig              `json:"monitoring"`
+	WebhookSecretName       string                        `json:"webhookSecretName"`
+	PodSpec                 *WrappedPodSpec               `json:"podSpec,omitempty"`
+	OperatorSecurityContext *corev1.SecurityContext       `json:"operatorSecurityContext,omitempty"`
```

Comment:
Rather than hanging this value off the main struct, can we just copy the convention set forth in the redpanda helmchart were we perform a pseudo merge patch? Doing so will be a bit more ergonomic and more importantly consistent across our charts.

Reply:
I removed `OperatorSecurityContext` and I took that value from `PodSpec`.